oak-run/src/main/java/org/apache/jackrabbit/oak/run/Downloader.java

```diff
@@ -101,16 +107,21 @@ public Downloader(int concurrency, int connectTimeoutMs, int readTimeoutMs, int
         this.bufferSize = bufferSize;
 
         this.executorService = new ThreadPoolExecutor(
-                (int) Math.ceil(concurrency * .1), concurrency, 60L, TimeUnit.SECONDS,
+                corePoolSize, concurrency, 60L, TimeUnit.SECONDS,
                 new LinkedBlockingQueue<>(),
                 BasicThreadFactory.builder().namingPattern("downloader-%d").daemon().build()
         );
+        this.executorServiceForParts = new ThreadPoolExecutor(
+                corePoolSize, concurrency, 60L, TimeUnit.SECONDS,
+                new LinkedBlockingQueue<>(),
+                BasicThreadFactory.builder().namingPattern("partDownloader-%d").daemon().build()
+        );
```

Comment:
by adding this thread pool executor, we are actually doubling the number of concurrent requests the downloader does. This might be confusing since `concurrency` is an option defined as you can see here.

```
OptionSpec<Integer> concurrencyOpt = parser.accepts("concurrency",
                        "Max number of concurrent requests that can occur (the default value is equal to 16 multiplied by the number of cores)")
                .withRequiredArg().ofType(Integer.class).defaultsTo(16 * Runtime.getRuntime().availableProcessors());
```

Reply:
Well, because we use the default LinkedBlockingQueue(), in reality each of the ThreadPoolExecutor will only create corePoolSize threads at most, and not more. All other tasks are waiting in the queue.

(We could replace LinkedBlockingQueue with eg. SynchronousQueue, but then we would get a RejectedExecutionException when we schedule more than max concurrency tasks... So that wouldn't simplify things).

I wanted to keep the changes to the minimum, so I didn't try to fix other things... So with 0.4 times concurrency corePoolSize, we have twice that number of threads, so still below the limit.